cart.asm

```diff
@@ -37,6 +37,7 @@ warmstart
     sta $ba ; last device
     lda #$1
     sta $8d
+    sta $de00 ; make sure we are in 16k mode (doesn't matter what we write, just a write does the trick)
```

Comment:
First and foremost, thanks for the contribution!

The [documentation](http://www.unusedino.de/ec64/technical/formats/crt.html) says, "...the upper area ($A000-$BFFF) may be disabled so Simons BASIC may use it as additional RAM. Writing a value of $01 to address location $DE00 banks in ROM, $00 disables ROM and enables RAM."

So the change makes sense. Maybe the comment could be updated to be more specific (I suppose "doesn't matter what we write" is not technically correct, as a 0 write would switch out ROM.)

Also, if you find the time, you could add to CHANGELOG.md "Unreleased" section.
Thank you!

Reply:
...hmmm... I just tried writing a 0 to $de00 here, but it does not seem to cause any problems at all, when I use the cartridge in VICE.
The cartridge boots fine, and I can start the "v" editor, enumerate the wordlist, et.c.

Is VICE wrong? It does not really make sense to me how it behaves.

Reply:
more weirdness in VICE. I start the cart, then "hex a000 dump" . it dumps the RAM contents. then "0 de00 c! hex a000 dump" dumps the ROM contents? It does not match the documentation on how $de00 should behave? I'm confused now.

Reply:
How VICE seems to actually behave:
Writing ANY number to $de00 switches in ROM, reading from $de00 switches in RAM.

I suppose the old documentation I have been following is wrong? In that case, your code (including comment) looks 100% correct.

Reply:
[https://groups.google.com/g/comp.sys.cbm/c/WLlpijJ8ZDk/m/sbCoQ_svN9IJ](url)

You can find the relevant source code for Vice in this file:

http://sourceforge.net/p/vice-emu/code/HEAD/tree/trunk/vice/src/c64/cart/simonsbasic.c

The comment at the beginning says:

- reading io1 switches to 8k game config (bank 0 at $8000)
- writing io1 switches to 16k game config (bank 0 at $8000, bank 1 at $a000)

Reply:
What I'm not sure is, how to know what is right, VICE or the old documentation. Anyway, for now it is practical to assume VICE is right :-)